delfin/drivers/hitachi/vsp/vsp_stor.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+
+import six
+from oslo_log import log
+from oslo_utils import units
+
+from delfin import exception
+from delfin.common import alert_util
+from delfin.common import constants
+from delfin.drivers import driver
+from delfin.drivers.hitachi.vsp import consts
+from delfin.drivers.hitachi.vsp import rest_handler
+
+LOG = log.getLogger(__name__)
+
+
+class HitachiVspDriver(driver.StorageDriver):
+    POOL_STATUS_MAP = {"POLN": constants.StoragePoolStatus.NORMAL,
+                       "POLF": constants.StoragePoolStatus.ABNORMAL,
+                       "POLS": constants.StoragePoolStatus.ABNORMAL,
+                       "POLE": constants.StoragePoolStatus.OFFLINE
+                       }
+    ALERT_LEVEL_MAP = {"Acute": constants.Severity.CRITICAL,
+                       "Serious": constants.Severity.MAJOR,
+                       "Moderate": constants.Severity.WARNING,
+                       "Service": constants.Severity.INFORMATIONAL
+                       }
+
+    TIME_PATTERN = '%Y-%m-%dT%H:%M:%S'
+
+    REFCODE_OID = '1.3.6.1.4.1.116.5.11.4.2.3'
+    DESC_OID = '1.3.6.1.4.1.116.5.11.4.2.7'
+    TRAP_TIME_OID = '1.3.6.1.4.1.116.5.11.4.2.6'
+    TRAP_DATE_OID = '1.3.6.1.4.1.116.5.11.4.2.5'
+    TRAP_NICKNAME_OID = '1.3.6.1.4.1.116.5.11.4.2.2'
+    LOCATION_OID = '1.3.6.1.4.1.116.5.11.4.2.4'
+    SECONDS_TO_MS = 1000
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.rest_handler = rest_handler.RestHandler(**kwargs)
+        self.rest_handler.login()
+
+    def reset_connection(self, context, **kwargs):
+        self.rest_handler.logout()
```

Comment:
verify should be updated before login

Reply:
changed

Reply:
Verified accessing with the certificate?

Reply:
not yet

Reply:
completed